portfolio/src/main/java/com/google/sps/servlets/UserInfoServlet.java

```diff
@@ -0,0 +1,52 @@
+package com.google.sps.servlets;
+
+import com.google.gson.*;
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.*;
+import java.io.IOException;
+
+@WebServlet("/userinfo")
+public class UserInfoServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Map<String, String> userInfo = new HashMap<String, String>();
+
+    // Gets user's info.
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      userInfo.put("isLoggedIn", "no");
```

Comment:
It should be a boolean type, otherwise you will encounter unexpected behaviour if the value is not "yes" or "no"